src/Bicep.Core.Samples/Files/InvalidModules_LF/main.diagnostics.bicep

```diff
@@ -70,7 +70,7 @@ module './main.bicep' = if
 
 module './main.bicep' = if (
 //@[7:21) [BCP096 (Error)] Expected a module identifier at this location. |'./main.bicep'|
-//@[28:28) [BCP009 (Error)] Expected a literal value, an array, an object, a parenthesized expression, or a function call at this location. ||
+//@[28:28) [BCP018 (Error)] Expected the ")" character at this location. ||
```

Comment:
>) [](start = 42, length = 1)

That's actually confusing. The original error message was correct.

Reply:
Will try to tweak the error recovery logic to restore the original error message.

Reply:
@majastrz I tried several ways and the one works best is tochange the type of `CloseParen` from `Token` to `SyntaxBase` in order to call `WithRecovery`. With this change the original error message is restored, and one bonus side effect is that we now have completions inside an un-closed parenthesized expression `(...` 😄

Reply:
Makes sense. The SyntaxBase approach is what we've done in other places too.

---
In reply to: [552174602](https://github.com/Azure/bicep/pull/1221#discussion_r552174602) [](ancestors = 552174602)